### PR TITLE
USHIFT-387: simplify configuration mechanisms

### PR DESCRIPTION
**Pull request by pacevedom**
**Which issue(s) this PR addresses**:

Closes #<Issue Number>

### CODE REVIEW

File: docs/howto_config.md [hunk 1]
```diff
@@ -1,11 +1,8 @@
 # Configuration
 
-MicroShift can be configured in the following ways, in order of precedence:
-* Command line arguments
-* Environment variables
-* Configuration file
+MicroShift can be configured via a configuration file only.
```
Comment (dhellmann) [inline docs/howto_config.md:3]: Saying "only" is relevant to someone who read the old docs, but doesn't feel necessary if we're just documenting how it works. Can we just remove this line, too, and not change the wording on what is now line 5?

**Comment by dhellmann** [comment]
I had one nit, but otherwise this looks good.

File: docs/howto_config.md (hunk 1 above)
Comment (pacevedom) [reply docs/howto_config.md]: Done

**Comment by zshi-redhat** [comment]
This change has an impact on the proposed pluggable enhancement which uses the env var mechnism for passing microshift config to plugin: https://github.com/openshift/enhancements/pull/1304/files#diff-3f76abf49288e379a81f4c6ba7e0724a6195dca9e5b90ee62663e29c9fa01242R180. 
Added a comment in the enhancement to update the design.

**Pull request merged**